components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -226,69 +227,155 @@ struct ActionGroup {
   array<ActionEntry> entries;
 };
 
-// Browser-side handler for requests from WebUI page.
-interface PageHandler {
-  SetClientPage(pending_remote<ChatUIPage> page);
-  // Get all the possible models for the conversation
+struct Conversation {
+  string uuid;
+  string title;
+  bool has_content;
+};
+
+interface Service {
+  // User opts-in to the feature at a profile level
+  MarkAgreementAccepted();
+
+  // Get metadata for non-archived conversations
+  GetVisibleConversations() => (array<Conversation> conversations);
+
+  // Get metadata for all quick actions
+  GetActionMenuList() => (array<ActionGroup> action_list);
+
+  // Current status of subscription
+  GetPremiumStatus() => (PremiumStatus status, PremiumInfo? info);
+
+  // Premium prompt is only shown conditionally (e.g. the user hasn't dismissed
+  // it and it's been some time since the user started using the feature).
+  GetCanShowPremiumPrompt() => (bool can_show);
+  DismissPremiumPrompt();
+
+  // Send events to the UI
+  BindObserver(pending_remote<ServiceObserver> ui);
+
+  // Bind specified Conversation for 2-way communication
+  BindConversation(
+      string conversation_id,
+      pending_receiver<ConversationHandler> conversation,
+      pending_remote<ConversationUI> conversation_ui);
+};
+
+interface ServiceObserver {
+  OnConversationListChanged(array<Conversation> conversations);
+  OnAgreementAccepted();
+};
+
+// Browser-side handler for general AI Chat UI functions, implemented
+// per-platform
+interface AIChatUIHandler {
+  // Open various external links or dialogs depending on the platform
+  OpenAIChatSettings();
+  OpenURL(url.mojom.Url url);
+  OpenLearnMoreAboutBraveSearchWithLeo();
+  OpenModelSupportUrl();
+  GoPremium();
+  RefreshPremiumSession();
+  ManagePremium();
+  HandleVoiceRecognition();
+
+  // This might be a no-op if the UI isn't closeable
+  CloseUI();
+
+  SetChatUI(pending_remote<ChatUI> chat_ui);
+
+  // Bind 2-way communication to the conversation related to the open page in
+  // the current browser window. No binding will occur if this isn't a
+  // tab-related UI, e.g. sidebar or popup.
+  BindRelatedConversation(
+      pending_receiver<ConversationHandler> conversation,
+      pending_remote<ConversationUI> conversation_ui);
+
+  // Make a completely new conversation, which may
+  // be bound to the associated conversation, if this UI has a target
+  NewConversation(pending_receiver<ConversationHandler> conversation,
+      pending_remote<ConversationUI> conversation_ui);
+
+  // TODO(petemill): WebUI clients won't need this when they can
+  // request via FaviconSource URLs via trusted webui.
+  GetFaviconImageData(string conversation_id) => (
```

Comment:
Yeah the comment is aiming to convey that we can't use `chrome://favicon[2]` because it's explicitly not available in Untrusted WebUI for security reasons.